### PR TITLE
Remove TitledDescription, tests and snapshots, and fix some compiler warnings

### PR DESCRIPTION
**Pull request by mluisbrown**
As has previously been discussed, it makes no sense for the `TitledDescription` component, which depends on ReactiveSwift, to continue as the only component in `BentoKit`, now that all the other components have been moved to Bento.

This PR removes `TitledDescription` from BentoKit.

### CODE REVIEW

**Review by diegopetrucci — APPROVED** [review]

**Review by NSMyself — APPROVED** [review]
👍

File: Bento/Components/Button.swift [hunk 1]
```diff
@@ -182,7 +182,7 @@ extension Component.Button {
 }
 
 public extension Component.Button {
-    public final class StyleSheet: InteractiveViewStyleSheet<View> {
+    final class StyleSheet: InteractiveViewStyleSheet<View> {
         public let button: ButtonStyleSheet
```
Comment (inamiy) [inline Bento/Components/Button.swift:186]: These `public`s will be confusing, so it's better to also remove all of them.

Comment (mluisbrown) [reply Bento/Components/Button.swift:186]: I just removed the ones that were causing compiler warnings.

Comment (inamiy) [reply Bento/Components/Button.swift:186]: Never mind, it's `public extension` which is already confusing... 🙄

**Pull request merged**